inferred-spans/src/test/java/co/elastic/otel/profiler/SamplingProfilerTest.java

```diff
@@ -275,6 +279,49 @@ void testVirtualThreadsExcluded() throws Exception {
     assertThat(profilingActive.get()).isFalse();
   }
 
+  @Test
+  void testTransactionWithRemoteParent() throws Exception {
+    setupProfiler(true);
+    awaitProfilerStarted(setup.profiler);
+
+    SpanContext dummyParentCtx =
+        SpanContext.createFromRemoteParent(
+            "a1a2a3a4a5a6a7a8b1b2b3b4b5b6b7b8",
+            "c1c2c3c4c5c6c7c8",
+            TraceFlags.getSampled(),
+            TraceState.getDefault());
+    Span remoteParent = Span.wrap(dummyParentCtx);
+    try (Scope scope = remoteParent.makeCurrent()) {
+      // ensure that a remote span activation does not trigger profiling
+      assertThat(setup.profiler.isProfilingActiveOnThread(Thread.currentThread())).isFalse();
+
+      Tracer tracer = setup.sdk.getTracer("manual-spans");
+      Span localRoot = tracer.spanBuilder("local-root").startSpan();
+      try (Scope scope2 = localRoot.makeCurrent()) {
+        Thread.sleep(500);
+      }
+      localRoot.end();
+    }
+
+    await()
+        .pollDelay(10, TimeUnit.MILLISECONDS)
+        .timeout(5000, TimeUnit.MILLISECONDS)
+        .untilAsserted(() -> assertThat(setup.getSpans()).hasSizeGreaterThanOrEqualTo(2));
+
+    List<SpanData> spans = setup.getSpans();
+    Optional<SpanData> localRoot =
+        spans.stream().filter(s -> s.getName().equals("local-root")).findAny();
+    assertThat(localRoot).isPresent();
+    assertThat(localRoot.get()).hasParentSpanId("c1c2c3c4c5c6c7c8");
+
+    assertThat(spans)
+        .anySatisfy(
+            span -> {
+              assertThat(span).hasParent(localRoot.get());
+              assertThat(span).hasAttribute(ElasticAttributes.IS_INFERRED, true);
```

Comment:
[minor] could we assert on the inferred span name so it's easier to understand that it's created from the call to `Thread.sleep` ?

Reply:
That's ufnortunately not possible without adding significant amounts of, because the inferred spans might or might not create an inferred span for the `testTransactionWithRemoteParent()` method depending on timing.